.github/workflows/merge.yml

```diff
@@ -0,0 +1,26 @@
+name: 'Merge to main'
+
+on:
+  workflow_dispatch: {}
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  build-and-deploy-documentation:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup
+        uses: ./.github/actions/setup
+      - name: Generate TypeDoc documentation
+        uses: ./.github/actions/build-docs
+      - name: Deploy documentation to gh-pages
+        uses: s0/git-publish-subdir-action@develop
```

Comment:
More information on this action can be found here: https://github.com/marketplace/actions/push-git-subdirectory-as-branch